packages/dev/serializers/src/USD/USDSerializer.ts

```diff
@@ -0,0 +1,731 @@
+
+        const transform = USDExport._BuildMatrix(mat);
+
+        if (mat.determinant() < 0) {
+            Tools.Warn(`USDExport: USDZ does not support negative scales on camera: ${camera.name}`);
+        }
+
+        if (camera.mode === Camera.ORTHOGRAPHIC_CAMERA) {
+            return `def Camera "${name}"
+                {
+                    matrix4d xformOp:transform = ${transform}
+                    uniform token[] xformOpOrder = ["xformOp:transform"]
+                    float2 clippingRange = (${camera.minZ.toPrecision(USDExport.Precision)}, ${camera.maxZ.toPrecision(USDExport.Precision)})
+                    float horizontalAperture = ${((Math.abs(camera.orthoLeft ?? 0) + Math.abs(camera.orthoRight ?? 0)) * 10).toPrecision(USDExport.Precision)}
+                    float verticalAperture = ${((Math.abs(camera.orthoTop ?? 0) + Math.abs(camera.orthoBottom ?? 0)) * 10).toPrecision(USDExport.Precision)}
+                    token projection = "orthographic"
+                }`;
+        } else {
+            return `def Camera "${name}"
+                {
+                    matrix4d xformOp:transform = ${transform}
+                    uniform token[] xformOpOrder = ["xformOp:transform"]
+                    float2 clippingRange = (${camera.minZ.toPrecision(USDExport.Precision)}, ${camera.maxZ.toPrecision(USDExport.Precision)})
+                    float focalLength = ${camera.fov.toPrecision(USDExport.Precision)}
+                    float focusDistance = ${USDExport.Focus.toPrecision(USDExport.Precision) /*THIS IS PROBABLY WRONG*/}
+                    float horizontalAperture = ${USDExport.FilmGauge.toPrecision(USDExport.Precision) /*THIS IS PROBABLY WRONG*/} 
+                    token projection = "perspective"
+                    float verticalAperture = ${USDExport.FilmGauge.toPrecision(USDExport.Precision) /*THIS IS PROBABLY WRONG*/}
+                }`;
+        }
+    }
+
+    private static async _BuildMaterials(materials: any, textures: any, quickLookCompatible: boolean = false) {
+        const array = [];
+        for (const uuid in materials) {
+            const material = materials[uuid];
+
+            array.push(await USDExport._BuildMaterial(material, textures, quickLookCompatible));
+        }
+
+        return `def "Materials"
+    {
+    ${array.join("")}
+    }
+    `;
+    }
+
+    private static async _BuildMaterial(_material: Material, textures: any, quickLookCompatible = false) {
+        // https://graphics.pixar.com/usd/docs/UsdPreviewSurface-Proposal.html
+        const pad = "			";
+        const inputs = [];
+        const samplers = [];
+        const material = _material as PBRMaterial;
+        const scene = material.getScene();
+
+        const channelIndexMap: { [key: string]: number } = {
+            r: 0,
+            g: 1,
+            b: 2,
+            a: 3,
+        };
+
+        async function createDynamicTextureFromChannel(texture: Texture, channel: string) {

```

Comment:
Why is this function not static? the rest of the class is all private/public static functions. This function doesn't use any reference to this, so it can be static as well, no? This way you don't need to create the class for not reason.

There are other functions like that.

I'm not the biggest fan of a class that only includes static functions, but if this was the selected architecture we can stick with it. Unless I missed something?

Reply:
I got you, and I guess I could define these all as flat functions then do 

```
function ExportAsBinaryZip(){
...
}

export const USDSerializer = {
 ExportAsBinaryZip
}
```

I just did it as a static class by preference, but if you don't like that I can change it!  I am getting to the end of nits that ill worry about for this PR though.  You have mentioned this twice now though so if you have an example you would like me to emulate let me know or ill do the above mentioned.

Reply:
My comment is more regarding consistency. Why are these functions class member, if the others are not? Class members mean that i need to construct a class to use them. And that they have "this" as context. But they don't use the context, they can be static too. Then the class is just a namespace, which is fine.
No need to remodel everything